### PR TITLE
PLC4X-57 Bugfix

### PR DESCRIPTION
**Pull request by JulianFeinauer**
Fixed PLC4X-57.
Also fixed Output from HelloPlc4x.java.
Fixed some small bugs.
Added TODO for returning arrays on bit streams and chars / strings.

### CODE REVIEW

**Comment by JulianFeinauer** [comment]
Sorry, the rebase did not work and I had to do it in a second commit.
@chrisdutz I added the functionality to return arrays in the Plx4XS7Protocol.java and it would be good if you look over it and my todo there.
Be aware, the code is pretty ugly due to some Type Erasure problems (Generic Arrays), especially in the bit-string case (but it was ugly there before so ... :D).

File: plc4j/protocols/s7/src/main/java/org/apache/plc4x/java/s7/netty/Plc4XS7Protocol.java [hunk 1]
```diff
@@ -420,22 +423,29 @@ private PlcResponse decodeReadResponse(S7ResponseMessage responseMessage, PlcReq
...
+                // TODO 2018-09-27 jf: array returning only implemented for BOOL, BYTE, INTEGERS, FP
+                // not for CHARS & STRINGS and not for all other bit-strings except for BYTE
                 switch (field.getDataType()) {
                     // -----------------------------------------
                     // Bit
                     // -----------------------------------------
                     case BOOL: {
-                        byte byteValue = data.readByte();
-                        fieldItem = new S7BooleanFieldItem(field.getDataType(),byteValue != 0x00);
+                        Boolean[] booleans = readAllValues(field, i -> data.readByte() != 0x00).toArray(new Boolean[field.getNumElements()]);
```
Comment (sruehl) [inline plc4j/protocols/s7/src/main/java/org/apache/plc4x/java/s7/netty/Plc4XS7Protocol.java]: actually sonar finds a presized array as a bug: see here https://stackoverflow.com/a/29444594/850036

Comment (JulianFeinauer) [reply plc4j/protocols/s7/src/main/java/org/apache/plc4x/java/s7/netty/Plc4XS7Protocol.java]: I'm about to fix it... didn't know that

**Comment by chrisdutz** [comment]
By the way ... looking good ... just had a few minutes after lunch and checked out your changes. 
And I already added a test for the field with array with a more than one digit number of elements in a separate test case.

**Pull request merged**